server/src/services/comment.service.ts

```diff
@@ -136,6 +139,13 @@ const CommentService = (
         payload: undefined,
       }
     }
+    if (comment.userId !== userId) {
```

Comment:
Como eu imaginei. A regra de negócio que o professor explicou é que só é possível apagar comentário e reação se o solicitante for o dono do post.